lib/src/integrationTest/kotlin/com/lemonappdev/konsist/api/ext/koscope/KoScopeExtTest.kt

```diff
@@ -0,0 +1,85 @@
+package com.lemonappdev.konsist.api.ext.koscope
+
+import com.lemonappdev.konsist.TestSnippetProvider
+import com.lemonappdev.konsist.api.declaration.KoBaseDeclaration
+import com.lemonappdev.konsist.api.declaration.KoClassDeclaration
+import com.lemonappdev.konsist.api.provider.KoBaseProvider
+import com.lemonappdev.konsist.api.provider.KoKDocProvider
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments.arguments
+import org.junit.jupiter.params.provider.MethodSource
+
+class KoScopeExtTest {
+    @ParameterizedTest
+    @MethodSource("provideValues")
+    fun `scope-contains-all-nested-and-local-base-declarations`(
+        includeNested: Boolean,
+        includeLocal: Boolean,
+    ) {
+        // given
+        val sut = getSnippetFile("scope-contains-all-nested-and-local-base-declarations")
+
+        // then
```

Comment:
Testing like this is ok?
```sut.declarationsOf<KoBaseDeclaration>(includeNested = includeNested, includeLocal = includeLocal)
            .shouldBeEqualTo(sut.declarations(includeNested, includeLocal))```
I mean I check not to a specific output (e.g. mapped to its names) but to `sut.declarations`

Reply:
These tests seem to be incorrect because they will not catch the issue if the original implementation of `declarations` breaks. Consider:
- adding JUnit test with mocks (desired)
- or testing this on a limited subset of data (comparing to original `declarations` and mentioning in a comment that declarations have a complete test suite)